x-pack/plugins/maps/common/descriptor_types/data_request_descriptor_types.ts

```diff
@@ -40,6 +40,7 @@ export type VectorSourceRequestMeta = DataFilters & {
   sourceQuery?: Query;
   sourceMeta: object | null;
   isForceRefresh: boolean;
+  editModeActiveForLayer?: boolean;
```

Comment:
You are calling this flag 2 different things in different places, `editModeActiveForLayer` and `isEditingFeatures`. How about just using a consistent value, something like `isFeatureEditorOpenForLayer`.
Which ever name is used, "feature edit" must be in it so its not confused with the layer edit functionallity.